src/main/frontend/handler/editor.cljs

```diff
@@ -70,26 +70,28 @@
     (when (:block/uuid block)
       (when-let [edit-id (state/get-edit-input-id)]
         (when-let [input (gdom/getElement edit-id)]
-          {:selection-start (util/get-selection-start input)
-           :selection-end (util/get-selection-end input)
-           :format (:block/format block)
-           :value (gobj/get input "value")
-           :block block
-           :edit-id edit-id
-           :input input})))))
-
-(defn- format-new-selection
-  [{:keys [selection-start selection-end value]}]
-  (let [selected (subs value selection-start selection-end)]
-    [(+ selection-start (count (take-while #(= " " %) selected)))
-     (- selection-end (count (take-while #(= " " %) (reverse selected))))]))
+          (let [selection-start (util/get-selection-start input)
+                selection-end (util/get-selection-end input)
+                value (gobj/get input "value")
+                selection (when (not= selection-start selection-end)
+                            (subs value selection-start selection-end))
+                selection-start (+ selection-start
+                                   (count (take-while #(= " " %) selection)))
+                selection-end (- selection-end
+                                 (count (take-while #(= " " %) (reverse selection))))]
+            {:selection-start selection-start
+             :selection-end selection-end
+             :selection selection
```

Comment:
Since the function name contains `get-selection`, a selected string or nil (if no selection) should be at least returned. 
In addition, it recalculates and returns the position of `selection-start` and `selection-end`, which was calculated by `format-new-selection` if the selection gets wrapped by whitespaces.